File: src/app/models/producto.ts
export class Producto {
   
          ProductoId : number; 
          DepartamentId : number;
          SubDepartamentId : number;
          ProductoNombre : string;
          ProductoDesc : string;
          MarcaId : number;
          PrecioCompra : number;
          PrecioBrutoVenta : number;
          PrecioNetoVenta  : number;
          ProductoImpuesto : number;
          ProductoImpuestoPorc : number;
          ProductoCantidad : number;

          constructor(){
              this.ProductoId = 0; 
                this.DepartamentId = 0;
                this.SubDepartamentId = 0;
                this.ProductoNombre = '';
                this.ProductoDesc = '';
                this.MarcaId = 0;
                this.PrecioCompra = 0;
                this.PrecioBrutoVenta = 0;
                this.PrecioNetoVenta  = 0;
                this.ProductoImpuesto = 0;
                this.ProductoImpuestoPorc = 0;
                this.ProductoCantidad = 0;
          }
}

File: Back_end/shopInlineApi/shopInlineApi/Models/Producto.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace shopInlineApi.Models
{
    public partial class Producto
    {
        public int ProductoId { get; set; }
        public int DepartamentId { get; set; }
        public int SubDepartamentId { get; set; }
        public string ProductoNombre { get; set; }
        public string ProductoDesc { get; set; }
        public int MarcaId { get; set; }
        public decimal PrecioCompra { get; set; }
        public decimal PrecioBrutoVenta { get; set; }
        public decimal PrecioNetoVenta { get; set; }
        public decimal ProductoImpuesto { get; set; }
        public decimal ProductoImpuestoPorc { get; set; }
        public int ProductoCantidad { get; set; }
        public DateTime CreateDate { get; set; }
        public DateTime ModifyDate { get; set; }

    public virtual Departement Departament { get; set; }
    public virtual Marca Marca { get; set; }
    public virtual SubDepartament SubDepartament { get; set; }
  }
}
